### ISSUE TITLE
Enhance components of textfield  with screen reader attributes

### ISSUE THREAD

**Issue by Larisa-Staroverova** [SAP/spartacus#14157]
Verify whether the screen reader reads everything as expected
Clarify what and how should be read with PO / UX
Enhance /add `aria` accordingly
